core/enum.go

```diff
@@ -163,6 +152,8 @@ func (e *ModuleEnum) DecodeInput(val any) (dagql.Input, error) {
 		return e.Lookup(x)
 	case dagql.Scalar[dagql.String]:
 		return e.Lookup(string(x.Value))
+	case *ModuleEnum:
+		return e.Lookup(x.Value)
```

Comment:
This is only slightly odd because no other scalar types support decoding from their own type, but I'm not gonna pretend it's wrong. IME there's not much downside to just letting these conversion functions grow organically in a system where we have to anticipate multiple representations of values. Definitely worth cleaning up that `switch` statement.

(Or: How I Learned to Stop Worrying and Love the Switch)